websites/A/AnimeLib/presence.ts

```diff
@@ -0,0 +1,348 @@
+import { AnimeLib } from "./lib";
+import type {
+	AnimeData,
+	UserData,
+	CharacterData,
+	PersonData,
+	CollectionData,
+	ReviewData,
+	TeamData,
+	PublisherData,
+} from "./lib";
+
+const presence = new Presence({
+		clientId: "1320289587943444552",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets { // Other default assets can be found at index.d.ts
+	Logo = "https://i.imgur.com/8vAN1DA.png",
+	Play = "https://cdn.rcd.gg/PreMiD/resources/play.png",
+	Pause = "https://cdn.rcd.gg/PreMiD/resources/pause.png",
+}
+
+interface IFrameVideo {
+	duration: number;
+	currentTime: number;
+	paused: boolean;
+}
+
+let iFrameVideo: IFrameVideo, startTimestamp: number, endTimestamp: number;
+
+presence.on("iFrameData", (data: IFrameVideo) => {
+	iFrameVideo = data;
+});
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+			type: ActivityType.Watching,
+			startTimestamp: browsingTimestamp,
+			smallImageText: "AnimeLIB",
+		},
+		isWatchingPrivately = await presence.getSetting<boolean>(
+			"private-watching"
+		),
+		path = document.location.pathname;
+
+	let animeData: AnimeData,
+		userData: UserData,
+		characterData: CharacterData,
+		peopleData: PersonData,
+		collectionData: CollectionData,
+		reviewData: ReviewData,
+		teamData: TeamData,
+		publisherData: PublisherData;
+
+	switch (`${path}/`.split("/")[2]) {
+		case "":
+			presenceData.details = "Главная страница";
+			presenceData.state = "Так внимательно изучает...";
+			break;
+		case "anime":
+			animeData = await AnimeLib.getAnime(
+				path,
+				path.split("/")[3].split("-")[0]
+			).then(response => <AnimeData>response.data);
+
+			// Show anime watching in private mode if it's enabled, or enforce it when anime is RX rated (e.g hentai)
+			if (isWatchingPrivately || animeData.ageRestriction.id === 5) {
+				presenceData.details = "Приватный режим";
+				presenceData.state = "Явно что-то скрывает...";
+
+				break;
+			}
+
+			if (animeData.toast) {
+				const cover =
+						document.querySelector<HTMLImageElement>(".cover__img")?.src,
+					title = document.querySelector("h1")?.textContent,
+					altTitle = document.querySelector("h2")?.textContent;
+
+				if (cover && title && altTitle) {
+					presenceData.details = "Страница лицензированного аниме";
+					presenceData.state = `${title} (${altTitle})`;
+					presenceData.largeImageKey = cover;
+				}
+
+				break;
+			}
+
+			if (path.endsWith("/watch")) {
+				const video = document.querySelector("video"),
+					dub = document
+						.querySelector(".menu-item.is-active")
+						?.querySelector(".menu-item__text").textContent;
+
+				if (dub) {
```

Comment:
What if there's no dub??

Reply:
If you mean what if there are only subtitles, then it'll be fine in any way, as the both menus (selecting dubs or subtitles) have same class names, and for the current selected one.
In case you mean if there are neither, then I doubt that such situation can ever happen on this site specifically.

Reply:
Okay good to know.